### PR TITLE
0-dimensions in mediapipe

### PR DESCRIPTION
**Pull request by dkalinowski**
- allow 0-dim during deserialization of KServe request to ov::Tensor/TF::Tensor
- block 0-dims in deserialization to MP::ImageFrame
- fix issue with missing buffer size check during TF tensor deserialization
- unit tests

JIRA: CVS-118193

### CODE REVIEW

**Review by michalkulakowski — APPROVED** [review]

File: WORKSPACE [hunk 1]
```diff
@@ -89,7 +89,7 @@ http_archive(
 git_repository(
     name = "mediapipe",
     remote = "https://github.com/openvinotoolkit/mediapipe",
-    commit = "c839182d40daae7270a29feba5bedd6330ccc2f8", # Cleanup and switch to tflite object detection (#16)
+    commit = "94dfb21bc60f82cd81c5839d3c68717f54edeedb", # TODO: dkalinow Pull Request
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply WORKSPACE]: This was required until I merge PR in our mediapipe fork. It is merged now, changed sha to latest main commit.

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 2]
```diff
@@ -124,21 +133,18 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
...
         if (expectedBytes != bufferLocation.size()) {
             std::stringstream ss;
             ss << "Expected: " << expectedBytes << " bytes; Actual: " << bufferLocation.size() << " bytes; input name: " << requestedName;
             const std::string details = ss.str();
             SPDLOG_DEBUG("[servable name: {} version: {}] Invalid content size of tensor proto - {}", request.model_name(), request.model_version(), details);
             return Status(StatusCode::INVALID_CONTENT_SIZE, details);
         }
-        outTensor = ov::Tensor(precision, shape, const_cast<void*>((const void*)bufferLocation.data()));
+        if (expectedBytes == 0) {
+            outTensor = ov::Tensor(precision, shape);  // OpenVINO does not accept nullptr as data ptr
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply src/mediapipe_internal/mediapipegraphexecutor.cpp:144]: Yes, the constructor used in line 144 creates tensor with tensor managed data (we do not pass data ptr). It allocates 0 bytes underneath.

File: src/rest_parser.cpp [hunk 3]
```diff
@@ -177,8 +177,25 @@ bool TFSRestParser::parseInstance(rapidjson::Value& doc) {
...
+            }
+            if (!addValue(proto, itr.value)) {
+                return false;
+            }
+        } else {
+            // If previous iterations already added instances (bateches)
+            // and those were non-arrays it means we have incorrect json
+            if (proto.tensor_shape().dim(0).size() > 1 &&
+                proto.tensor_shape().dim_size() == 1) {
+                return false;
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply src/rest_parser.cpp]: It's not possible to send 1-d array using row format. The format allows sending batches of "something".
"inputs": [1,2,3]} is column format, so it is not relevant, but:
"instances": [1,2,3]} is of shape (1,3) (1 batch of data [1,2,3])

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 4]
```diff
@@ -90,13 +90,22 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
         TensorShape::BuildTensorShapeBase(rawShape, static_cast<tensorflow::TensorShapeBase<TensorShape>*>(&tensorShape));
         // TODO here we allocate default TF CPU allocator
         tensorflow::Tensor localTensor(datatype, tensorShape);  // TODO error handling
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply src/mediapipe_internal/mediapipegraphexecutor.cpp]: done

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 5]
```diff
@@ -208,21 +214,35 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
...
+        std::stringstream ss;
+        ss << "Invalid Mediapipe Image input number of channels. Expected greater than 0; Actual: " << numberOfChannels << "; Expected layout - HWC.";
+        const std::string details = ss.str();
+        SPDLOG_DEBUG(details);
+        return Status(StatusCode::INVALID_SHAPE, details);
+    }
+    int64_t numberOfRows = requestInputItr->shape()[0];
+    if (numberOfRows <= 0) {
+        std::stringstream ss;
+        ss << "Invalid Mediapipe Image input height. Expected greater than 0; Actual: " << numberOfRows << "; Expected layout - HWC.";
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply src/mediapipe_internal/mediapipegraphexecutor.cpp:225]: done

**Review by mzegla — APPROVED** [review]

**Pull request merged**